packages/service-library/tests/rabbitmq/test_rabbitmq_rpc_router.py

```diff
@@ -0,0 +1,94 @@
+# pylint:disable=redefined-outer-name
+# pylint:disable=unused-argument
+
+from collections.abc import Awaitable, Callable
+
+import orjson
+import pytest
+from faker import Faker
+from servicelib.rabbitmq import RabbitMQClient
+from servicelib.rabbitmq_rpc_router import RPCRouter
+from servicelib.rabbitmq_utils import RPCMethodName, RPCNamespace
+
+pytest_simcore_core_services_selection = [
+    "rabbit",
+]
+
+
+@pytest.fixture
+def rabbitmq_rpc_client(
+    rabbitmq_client: Callable[[str], RabbitMQClient]
+) -> Callable[[str], Awaitable[RabbitMQClient]]:
+    async def _creator(name: str) -> RabbitMQClient:
+        rpc_client = rabbitmq_client(name)
+        assert rpc_client
+        await rpc_client.rpc_initialize()
+        return rpc_client
+
+    return _creator
+
+
+router = RPCRouter()
+
+
+@router.expose()
+async def a_str_method(
+    a_global_arg: str, *, a_global_kwarg: str, a_specific_kwarg: str
+) -> str:
+    return f"{a_global_arg}, that was a winner! {a_global_kwarg} {a_specific_kwarg}"
+
+
+@router.expose()
+async def an_int_method(a_global_arg: str, *, a_global_kwarg: str) -> int:
+    return 34
+
+
+@router.expose()
+async def a_raising_method(a_global_arg: str, *, a_global_kwarg: str) -> int:
+    msg = "This is not good!"
+    raise RuntimeError(msg)
+
+
+@pytest.fixture
+def router_namespace(faker: Faker) -> RPCNamespace:
+    return faker.pystr()
+
+
+async def test_exposed_methods(
+    rabbitmq_rpc_client: Callable[[str], Awaitable[RabbitMQClient]],
+    router_namespace: RPCNamespace,
+    cleanup_check_rabbitmq_server_has_no_errors: None,
+):
+    rpc_client = await rabbitmq_rpc_client("client")
```

Comment:
Same here. I think the design should limit functions exposed for `rpc_server` and those by `rpc_client`